### PR TITLE
[WIP] Deprecate pyspark-notebook, r-notebook, Mesos support

### PR DESCRIPTION
**Pull request by parente**
* Removes the pyspark-notebook and r-notebook image definitions based on usage stats https://nbviewer.jupyter.org/gist/parente/316d5c242aeb484484c8 (#517, #577)
* Removes the Mesos client libraries from the all-spark-notebook as a step toward making the image Hadoop and cluster manager agnostic (#626, #663)
* Updates documentation

It's not meant to be the one and only PR for #626, only a step in that direction.

### CODE REVIEW

**Comment by doctapp** [comment]
Why remove based on stats? We extensively use pyspark-notebook by modding it and only pulling it's base image (scipy-notebook)... Also, how will Mesos be supported if deprecating it?
Thanks

**Comment by parente** [comment]
Thanks for the input @doctapp. 

> Why remove based on stats? 

Mainly to cut down on build time and maintenance cost here.

> How will Mesos be supported if deprecating it?

It wouldn't be supported here, since we (the current maintainers) have no way to test it against a Mesos cluster. We're thinking of making all-spark-notebook a cluster manager agnostic image, and rely on someone with Mesos experience and interest in the community to build a descendant image or one that could be used in conjunction with it. 

The issues linked in the description have some discussion on these topics.

**Comment by doctapp** [comment]
Understood, thanks.

File: docs/using/specifics.md [hunk 1]
```diff
@@ -58,111 +58,15 @@ val rdd = sc.parallelize(0 to 999)
...
-
-```scala
-// should print the value of --master in the kernel spec
-println(sc.master)
-
-// do something to prove it works
-val rdd = sc.parallelize(0 to 99999999)
-rdd.sum()
-```
-
```
Comment (taqtiqa-mark) [inline docs/using/specifics.md]: Maybe this content in `docs/using/specifics.md` can be moved to `examples/mesos.md`?

Comment (parente) [reply docs/using/specifics.md]: That's an option. Another is to move them out entirely. One reason for deprecating these cluster-specific notes is that we don't have a way to maintain them.

Comment (taqtiqa-mark) [reply docs/using/specifics.md]: Ack  No problem with deleting.  Just seemed a waste to dispose of all the effort someone put into learning.  Anyway likely better suited to the Docker-recipes wiki page referenced in a comment to #626.

**Comment by dungdm93** [comment]
If pyspark-notebook, r-notebook are deprecated, should we simplify `all-spark-notebook` name to `spark-notebook`?

**Comment by romainx** [comment]
Hello,

Just to share some thoughts about this. It makes sense since.

- `r-notebook` could be replaced without loss of features by `datascience-notebook`
- `pyspark-notebook` could be replaced without loss of features by `all-spark-notebook`

If this decision is taken, we could

- Put a deprecation notice in the documentation for these two images -> I can take care of this point if you want.
- Maybe move them to community images in order to give to users a chance to continue using them.

Best.

**Comment by parente** [comment]
I'm going to close this PR as irrelevant.

* We removed the mesos clients in another PR.
* We haven't hit issues exceeding CI time limits due to the number of images in quite some time.
* We're not running into issues hosting all of the existing images on Docker Hub.
* It's not terribly difficult to maintain the r-notebook and pyspark-notebook alongside the others.
* We already closed lll of the issues linked from the description.

If conditions change, we can reconsider. Right now, there's little benefit to doing this work and only disruption for users of the two images.

**Pull request closed without merging**